rust_src/src/frames.rs

```diff
@@ -593,4 +607,26 @@ pub fn frame_focus(frame: LispFrameLiveOrSelected) -> LispObject {
     frame_ref.focus_frame
 }
 
+/// Return a list of all live frames.
+/// The return value does not include any tooltip frame.
+#[lisp_fn]
+pub fn frame_list() -> LispObject {
+    #[cfg(feature = "window-system")]
+    {
+        let mut list: LispObject = Qnil;
+        for_each_frame!(f => {
+            if !f.has_tooltip() {
+                    list = LispObject::cons(LispObject::from(f), list);
```

Comment:
This is no longer the correct style.

`list = (f, list).into();` should be all you need.